internal/pkg/util/fs/helper.go

```diff
@@ -14,9 +14,36 @@ import (
 	"strings"
 	"syscall"
 
+	"github.com/sylabs/singularity/internal/pkg/sylog"
 	"golang.org/x/sys/unix"
 )
 
+// EnsureFilePrivate takes a file path, and 1. Creates it with
+// the specified permission, or 2. ensures a file path is the
+// specified permission.
+func EnsureFilePermission(fn string, mode os.FileMode) error {
```

Comment:
I'd prefer something without the side effect of creating an empty file tbh. If this ensures a file is present with certain permission maybe `EnsureFileAndPermission` is a better name for what it's doing.

Also, please could you add a test to `helper_linux_test.go` for things added here - to keep up the nice coverage on these helpers.

Reply:
Okay, I will change it to `EnsureFileAndPermission`

Reply:
or `EnsureFileWithPermission` ... is possibly nicer ... up to you :-)